core/src/main/java/org/elasticsearch/discovery/zen/elect/ElectMasterService.java

```diff
@@ -36,13 +37,15 @@
 
     public static final String DISCOVERY_ZEN_MINIMUM_MASTER_NODES = "discovery.zen.minimum_master_nodes";
 
+    private final Version version;
     private final NodeComparator nodeComparator = new NodeComparator();
 
     private volatile int minimumMasterNodes;
 
     @Inject
-    public ElectMasterService(Settings settings) {
+    public ElectMasterService(Settings settings, Version version) {
```

Comment:
can we rename version into minMasterVersion, call it with Version. minimumCompatibilityVersion() and document what it means? I think it will be clearer.